api/app/controllers/spree/api/v2/storefront/stores_controller.rb

```diff
@@ -9,7 +9,9 @@ def current
 
           def export_google_rss_feed
             settings
-            if @settings.enabled
+            if @settings.nil?
+              raise ActionController::RoutingError, 'Not Found'
```

Comment:
why you raising `ActionController::RoutingError` if we expecting that `@settings` will be AR instance ? Shouldn't be `ActiveRecord::RecordNotFound` ?

Reply:
I think a regular 404. If you call `find!` (raises NotFoundError https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by-21), the exception will be picked up by Rails on its own and translated into a 404 error.